### PR TITLE
 [irrlicht] Add OSX support

### PR DESCRIPTION
**Pull request by shybovycha**
Add missing (or rather violently manually disabled) support for OSX host systems, since Irrlicht natively supports OSX, being a cross-platform 3D engine.

- #### What does your PR fix?  

Add missing platform support.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

Tested successfully on OSX Big Sur

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  

Yes

### CODE REVIEW

File: ports/irrlicht/CONTROL [hunk 1]
```diff
@@ -1,9 +1,9 @@
 Source: irrlicht
 Version: 1.8.4-10
```
Comment (NancyLi1013) [inline ports/irrlicht/CONTROL:2]:
```suggestion
Version: 1.8.4
Port-version: 11
```

File: ports/irrlicht/portfile.cmake [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+vcpkg_fail_port_install(ON_ARCH "arm" ON_TARGET "uwp")
 
 vcpkg_from_sourceforge(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO irrlicht/Irrlicht%20SDK
     REF 1.8/1.8.4
     FILENAME "irrlicht-1.8.4.zip"
     SHA512 de69ddd2c6bc80a1b27b9a620e3697b1baa552f24c7d624076d471f3aecd9b15f71dce3b640811e6ece20f49b57688d428e3503936a7926b3e3b0cc696af98d1
-    PATCHES fix-encoding.patch fix-sysctl.patch
+    PATCHES fix-encoding.patch fix-sysctl.patch fix-osx-compilation.patch
```
Comment (NancyLi1013) [inline ports/irrlicht/portfile.cmake:9]:
```suggestion
    PATCHES 
        fix-encoding.patch 
        fix-sysctl.patch 
        fix-osx-compilation.patch
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Could you please remove `irrlicht:x64-osx=fail` in ci.baseline.txt?

https://github.com/microsoft/vcpkg/blob/d2717c38db071506b13f9b0afb41a96e92bf6959/scripts/ci.baseline.txt#L490

Seems you forgot to update this file https://github.com/microsoft/vcpkg/blob/master/versions/baseline.json.

File: versions/i-/irrlicht.json [hunk 3]
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "e2b2dd836e693401dab067c5064665e926a46b07",
+      "git-tree": "18ae08d6099b5457ccedbdb7209c79321a9e9027",
+      "version-string": "1.8.4",
+      "port-version": 11
+    },
+    {
+      "git-tree": "da996239bd0a673f57f59686d2fdf350e9b5ea64",
```
Comment (NancyLi1013) [inline versions/i-/irrlicht.json:9]: This should not be changed. You can run `.\vcpkg x-add-version --overwrite-version simage` to update this.
```suggestion
      "git-tree": "e2b2dd836e693401dab067c5064665e926a46b07",
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Related issue #15011

File: versions/i-/irrlicht.json (hunk 3 above)
Comment (shybovycha) [reply versions/i-/irrlicht.json:9]: i am not entirely sure why this should *not* change - i do not update the versions manually, just running `vcpkg x-add-version --overwrite-version irrlicht` **after the commit**.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Have you tested these features in this PR?

**Comment by shybovycha** [comment]
Yes, as I have mentioned in the PR description, OSX Big Sur ✔
Additionally, just tested on Windows 10, ✔

**Comment by NancyLi1013** [comment]
`qt5-3d` failed on x64-osx:

`ERROR: Feature 'system-assimp' was enabled, but the pre-condition 'features.assimp && libs.assimp' failed.`

Could you please look into this?

**Comment by shybovycha** [comment]
I saw `qt5-3d` failing yesterday, thought it was some infrastructure issue - it has nothing to do with Irrlicht, really...

**Comment by shybovycha** [comment]
Okay, figured it out: the issue is in the `qt5-3d` port itself - it [carelessly tries to link](https://github.com/microsoft/vcpkg/blob/master/ports/qt5-3d/portfile.cmake#L46-L50) `-lGL -lX11 -lXxf86vm` on non-windows platforms. This fails on OSX.

```
if(VCPKG_TARGET_IS_WINDOWS)
    set(SYSTEM_LIBS "Advapi32.lib user32.lib gdi32.lib")
else()
    set(SYSTEM_LIBS "-lGL -lXxf86vm -lX11")
endif()
```

Hence adding the patch for that one too:

```
if(VCPKG_TARGET_IS_WINDOWS)
    set(SYSTEM_LIBS "Advapi32.lib user32.lib gdi32.lib")
elseif(VCPKG_TARGET_IS_OSX)
    set(SYSTEM_LIBS "-framework OpenGL -framework IOKit -framework Cocoa -framework Carbon -framework AppKit")
else()
    set(SYSTEM_LIBS "-lGL -lXxf86vm -lX11")
endif()
```

**Review by NancyLi1013 — COMMENTED** [review]
Could you please also add `Port-Version: 1` in CONTROL file for `qt5-3d`?

https://github.com/microsoft/vcpkg/blob/640c439df78158904992c51fcbc46f64da113690/ports/qt5-3d/CONTROL#L2

```
Version: 5.15.2
Port-Version: 1
```

**Comment by shybovycha** [comment]
Makes sense, but this change added a new version to the `versions/q-/qt5-3d.json` - should there be just one?

File: versions/q-/qt5-3d.json [hunk 4]
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "c9a8b7a2360c554409f283143edf9202bebe6a3b",
+      "git-tree": "6c3c15324a4764d3501798299c8ad8087905b136",
+      "version-string": "5.15.2",
+      "port-version": 1
+    },
+    {
+      "git-tree": "485d3bc3da70ca095891f8641bafd228e1538232",
```
Comment (NancyLi1013) [inline versions/q-/qt5-3d.json:9]: This parts seem incorrect. Do you run `.\vcpkg x-add-version qt5-3d` to change this? It should be only new one changed like `irrlicht`.

To make sure all look fine, you can run `.\vcpkg x-add-version --overwrite-version --all` to update, then commit the changes.

**Comment by NancyLi1013** [comment]
> Makes sense, but this change added a new version to the `versions/q-/qt5-3d.json` - should there be just one?

Yes, there should be just new one added.

Note: Sorry for my wrong operation. I should add comment in a new line instead of editing your content.

File: versions/q-/qt5-3d.json (hunk 4 above)
Comment (shybovycha) [reply versions/q-/qt5-3d.json:9]: that is weird - `vcpkg x-add-version qt5-3d --overwrite-version` is exactly what produced this code... you reckon this is a bug in vcpkg itself?..

Comment (NancyLi1013) [reply versions/q-/qt5-3d.json:9]: You changed this file manually last time? Please revert this change in versions/q-/qt5-3d.json first, and then `vcpkg x-add-version qt5-3d`.

Comment (shybovycha) [reply versions/q-/qt5-3d.json:9]: No, that's the thing: i never changed version files manually - only by the `vcpkg x-ad-version --overwrite-version`

Comment (NancyLi1013) [reply versions/q-/qt5-3d.json:9]: It's so weird. I also don't know what happened here. I checked that this commit seems normal https://github.com/microsoft/vcpkg/pull/18725/commits/f234b5f8996e7a70389898aefc5a5e12dfe20960.

Comment (NancyLi1013) [reply versions/q-/qt5-3d.json:9]: Now the value for `git-tree` is correct, the format changed. Not clear why this changed.

**Review by ras0219-msft — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your fixing @shybovycha.

**Pull request merged**